website/static/js/profile.js

```diff
@@ -468,19 +479,37 @@ var SocialViewModel = function(urls, modes) {
     var self = this;
     BaseViewModel.call(self, urls, modes);
     TrackedMixin.call(self);
-
+    var i;
     self.addons = ko.observableArray();
 
-    self.personal = extendLink(
-        // Note: Apply extenders in reverse order so that `ensureHttp` is
-        // applied before `url`.
-        ko.observable().extend({
-            trimmed: true,
-            url: true,
-            ensureHttp: true
-        }),
-        self, 'personal'
-    );
+    self.canEditWebsite = ko.observable(false);
+
+    self.editWebsiteButton =  function(){
+        self.canEditWebsite(!self.canEditWebsite());
+    }
+
+    self.profileWebsites = ko.observableArray();
+                
+    self.hasProfileWebsites = ko.computed(function() {
+        if (self.profileWebsites())      
+        for (var i=0; i<self.profileWebsites().length; i++) {
+            if (self.profileWebsites()[i]) {
+                return true;
+            }
+        }
+        return false;
+    });
+    
+    self.canRemove = ko.computed(function () {
+        if (self.profileWebsites()) {
+            return self.profileWebsites().length > 1;
```

Comment:
canRemove is used to tell if we should remove the dialog box when you hit the remove button, or just clear the field.  So when you hit remove on the last website, it does not look like this.  

![image](https://cloud.githubusercontent.com/assets/6232068/7572805/f0ab06f4-f7ee-11e4-8413-c3075ec18711.png)

The blank sites are stripped before uploading to the server.